### PR TITLE
TC03-User cannot log into Railway with invalid password; TC04-Login page displays when un-logged User clicks on "Book ticket" tab; TC06-Additional pages display once user logged in

### PR DESCRIPTION
**Pull request by sattdn210106**

### CODE REVIEW

File: src/test/java/tests/LoginTest.java [hunk 1]
```diff
@@ -35,4 +36,53 @@ public void TC002() {
...
+        homePage.gotoLoginPage();
+
+        String invalidPassword = DataHelper.getRandomText();
+
+        loginPage.login(Constant.USERNAME, invalidPassword);
+
+        String expectedErrorMsg = "There was a problem with your login and/or errors exist in your form.";
+        String actualErrorMsg = loginPage.getErrorMsg();
+
+        Assert.assertEquals(actualErrorMsg, expectedErrorMsg, "Error message is incorrect.");
```
Comment (thanhlelgg) [inline src/test/java/tests/LoginTest.java:53]: why two verify point with only different failed message?

File: src/test/java/tests/LoginTest.java [hunk 2]
```diff
@@ -35,4 +36,53 @@ public void TC002() {
...
+        loginPage.login(Constant.USERNAME, Constant.PASSWORD);
+
+        Assert.assertTrue(homePage.doesTabMyTicketExist());
+        Assert.assertTrue(homePage.doesTabLogoutExist());
+        Assert.assertTrue(homePage.doesChangePasswordExist());
+
+        homePage.gotoMyTicketPage();
+
+        String expectedMyTicketPageTitle = "Safe Railway - My Ticket";
+        String actualPageTitle = BrowserHelper.getPageTitle();
```
Comment (thanhlelgg) [inline src/test/java/tests/LoginTest.java]: you should not use page title to verify if you are at a page if there's better option like page header, breadcrumb or sth like that.
when we check if we are at a page, we care more about the UI, and less about browser tab title

File: src/test/java/tests/LoginTest.java [hunk 3]
```diff
@@ -35,4 +36,53 @@ public void TC002() {
...
+
+        Assert.assertEquals(actualPageTitle, expectedMyTicketPageTitle, "My Ticket page doesn't display");
+
+        homePage.gotoChangePasswordPage();
+
+        String expectedChangePasswordPageTitle = "Safe Railway - Change Password";
+        actualPageTitle = BrowserHelper.getPageTitle();
```
Comment (thanhlelgg) [inline src/test/java/tests/LoginTest.java]: same as above

**Review by thanhlelgg — CHANGES_REQUESTED** [review]

File: src/test/java/tests/LoginTest.java (hunk 1 above)
Comment (sattdn210106) [reply src/test/java/tests/LoginTest.java:53]: Thanks for your review. I just have updated my code, please review it for me.

File: src/test/java/tests/LoginTest.java (hunk 2 above)
Comment (sattdn210106) [reply src/test/java/tests/LoginTest.java]: Thanks for your review. I just have updated my code, please review it for me.

File: src/test/java/tests/LoginTest.java (hunk 3 above)
Comment (sattdn210106) [reply src/test/java/tests/LoginTest.java]: Thanks for your review. I just have updated my code, please review it for me.

**Review by thanhlelgg — APPROVED** [review]

**Pull request merged**